src/proto/message_translation/power_frame_msg.cpp

```diff
@@ -0,0 +1,161 @@
+#include <pb_encode.h>
+#include <proto/primitive.nanopb.h>
+
+#ifdef PLATFORMIO_BUILD
+#include <power_frame_msg_platformio.h>
+#else
+#include "pb_decode.h"
+#include "pb_encode.h"
+#include "proto/message_translation/power_frame_msg.h"
+
+#endif
+
+std::vector<uint8_t> serializeToVector(const TbotsProto_PowerFrame& data)
+{
+    std::vector<uint8_t> buffer(TbotsProto_PowerFrame_size);
+    pb_ostream_t stream =
+        pb_ostream_from_buffer(static_cast<uint8_t*>(buffer.data()), buffer.size());
+    if (!pb_encode(&stream, TbotsProto_PowerFrame_fields, &data))
+    {
+        throw std::runtime_error(
+            "Failed to encode PowerFrame msg to vector when converting nanopb to vector");
+    }
+    return buffer;
+}
+
+std::vector<uint8_t> serializeToVector(const TbotsProto_PowerControl& data)
+{
+    std::vector<uint8_t> buffer(TbotsProto_PowerControl_size);
+    pb_ostream_t stream =
+        pb_ostream_from_buffer(static_cast<uint8_t*>(buffer.data()), buffer.size());
+    if (!pb_encode(&stream, TbotsProto_PowerControl_fields, &data))
+    {
+        throw std::runtime_error(
+            "Failed to encode PowerControl msg to vector when converting nanopb to vector");
+    }
+    return buffer;
+}
+
+std::vector<uint8_t> serializeToVector(const TbotsProto_PowerStatus& data)
+{
+    std::vector<uint8_t> buffer(TbotsProto_PowerStatus_size);
+    pb_ostream_t stream =
+        pb_ostream_from_buffer(static_cast<uint8_t*>(buffer.data()), buffer.size());
+    if (!pb_encode(&stream, TbotsProto_PowerStatus_fields, &data))
+    {
+        throw std::runtime_error(
+            "Failed to encode PowerStatus msg to vector when converting nanopb to vector");
+    }
+    return buffer;
+}
+
+void setPowerMsg(TbotsProto_PowerFrame& frame, const TbotsProto_PowerControl& control)
+{
+    frame.which_power_msg         = TbotsProto_PowerFrame_power_control_tag;
+    frame.power_msg.power_control = control;
+}
+
+void setPowerMsg(TbotsProto_PowerFrame& frame, const TbotsProto_PowerStatus& status)
+{
+    frame.which_power_msg        = TbotsProto_PowerFrame_power_status_tag;
+    frame.power_msg.power_status = status;
+}
```

Comment:
Do we need these functions? I think if you took the tags as an argument,  we might be able to get rid of these functions

Reply:
Theyre more so to clean up uart_framing.hpp. The alternative I thought of was using checking TYPENAME(type)

Reply:
This won't be possible due to needing `if constexpr (...` because we need to set either frame.power_msg.power_control or frame.power_msg.power_status